internal/build/lifecycle_execution.go

```diff
@@ -706,6 +744,133 @@ func (l *LifecycleExecution) ExtendRun(ctx context.Context, buildCache Cache, ph
 	defer extend.Cleanup()
 	return extend.Run(ctx)
 }
+func (l *LifecycleExecution) ExtendBuildByDaemon(ctx context.Context) error {
+	extendtime := time.Now()
+	builderImageName := l.opts.BuilderImage
+	defaultFilterFunc := func(file string) bool { return true }
+	var extensions Extensions
+	time1 := time.Now()
+	extensions.SetExtensions(l.tmpDir, l.logger)
+	l.logger.Debugf("extensions.SetExtensions for build took %s", time.Since(time1))
+	time2 := time.Now()
+	dockerfiles, err := extensions.DockerFiles(DockerfileKindBuild, l.tmpDir, l.logger)
+	if err != nil {
+		return fmt.Errorf("getting %s.Dockerfiles: %w", DockerfileKindBuild, err)
+	}
+	l.logger.Debugf("extensions.DockerFiles for build took %s", time.Since(time2))
+	dockerapplytime := time.Now()
+	for _, dockerfile := range dockerfiles {
+		buildContext := archive.ReadDirAsTar(filepath.Dir(dockerfile.Path), "/", 0, 0, -1, true, false, defaultFilterFunc)
```

Comment:
The build context should be the application workspace, see here: https://github.com/buildpacks/lifecycle/blob/e06b69b5acb14e441659178b13de4a38e7a2e612/internal/extend/kaniko/dockerfile_applier.go#L88